### ISSUE TITLE
Some code issue

### ISSUE THREAD

**Issue by Harry-PARK** [hehuannb/FlexGen-EHR#2]
### **main.py**

Change MIMICDATASET class. (I take this code from run_vae.py)
-----------------------------------------------------------------------
    class MIMICDATASET(Dataset):
        def __init__(self, x_t,x_s, y, train=None, transform=None):
            # Transform
            self.transform = transform
            self.train = train
            self.xt = x_t
            self.xs = x_s
            self.y = y
  
        def return_data(self):
            return self.xt, self.xs, self.label
    
        def __len__(self):
            return len(self.xt)
    
        def __getitem__(self, idx):
            sample = self.xt[idx]
            stat = self.xs[idx]
            sample_y = self.y[idx]
            return sample, stat, sample_y

B style is better when changing a task. 
-----------------------------------------------------------------------
    y = torch.tensor(df_pop.ARF_LABEL.values).to(torch.float32)    # A
    y = torch.tensor(df_pop["ARF_LABEL"].values).to(torch.float32)    # B

Deleted wrong code. I think this is not necessary.
Instead, I refer to run_vae.py code 
-----------------------------------------------------------------------
    ys, yt = y[idx_s, idx_t]    # wrong code
    # replaced with codes below
    y = torch.tensor(df_pop["mortality_LABEL"].values).to(torch.float32)
    print(y.shape)
    dataset_train_object = MIMICDATASET(x_t, x_s, y,  train=True, transform=False)
    train_loader = DataLoader(dataset_train_object, batch_size=batch_size, shuffle=True, num_workers=1, drop_last=False)

### **flexgen.py**

changed encode() and diff() parameters in flexgen.train_epoch()
-----------------------------------------------------------------------
    xt = xt.to(device)
    xs = xs.to(device)
    c = c.to(device)
 
    ms, ss = self.svae.encode(xs, c) # 수정
    zs = self.svae.reparameterize(ms, ss)
    mt, st = self.tvae.encode(xt, c) # 수정
    zt = self.tvae.reparameterize(mt, st)
    z = torch.concat([zt, zs],dim=1)
  
    loss_d = self.diff(z, c)

changed deocde part in flexgen.generate()
-----------------------------------------------------------------------
    label = torch.tensor(label).to(device).unsqueeze(0).repeat(num_sample)

    xt_gen = self.tvae.decode(z_t, label)
    xs_gen = self.svae.decode(z_s, label)

### **torch.save/torch.load**
changed to save and load weights only. (This is to prevent to use depreciated code and secure stability)
-----------------------------------------------------------------------
    For example)
    torch.save(net.state_dict(), 'saved_models/vae_stat.pth')
    ...
    svae = VariationalAutoencoder(feature_dim_s).to(device)
    tvae_dict = torch.load('saved_models/vae_tmp.pth', weights_only=True)
    tvae.load_state_dict(tvae_dict)
